### ISSUE TITLE
Simplify debug

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode-cpptools#8224]
Hi VS Code PM here 👋 

I just tried setting up C++ with VS Code following [this tutorial](https://code.visualstudio.com/docs/cpp/config-clang-mac). The flow is actually nice, but here's some feedback on what I think we could improve:

1. C++ should consider contributing to the Editor title area menu, such that a Play button would appear in the Editor Title area. [Example how to do this](https://github.com/microsoft/vscode-mock-debug/blob/main/package.json#L80). I am not sure if it would be too noisy to have this on every file C++.
2. You aggressively generate `tasks.json` and `launch.json` and you do not provide configurations from memory. Consider providing a dynamic `resolveDebugConfiguartion` that will provide a launch configuration on the fly. [Example](https://github.com/microsoft/vscode-mock-debug/blob/main/src/activateMockDebug.ts#L161)
3. If you decide to stick with `launch.json` generation consider to simplify it. For example `stopOnEntry` field could be removed.
4. If I have a purple no folder VS Code it is impossible to debug a simple C++ file

I would love if we could improve this experience, since we see in user studies that users want a confirmation that VS Code can do C++, and providing an easy way to run and debug simple C++ files would be great for this.

Let me know if I can help somehow. I am also open to a quick 30 min meeting where we can discuss about the new cool VS Code debug features that C++ could potentially adopt. And you could let us know what are your biggest asks from the VS Code debugger.

fyi @WardenGnaw @weinand

**Comment by WardenGnaw** [comment]
@jureid Is working on a design document similar to this issue.

**Comment by bobbrow** [comment]
@jureid is this covered by #7497?

**Comment by jureid** [comment]
Yup! I chatted with Ela last week about supporting the purple no folder experience with the Play button. I think she needed to figure out how to test that scenario... I'll follow up with her this week to see if she needs any more spec work for that.
